### ISSUE TITLE
Correctly support desiredSize and minSize of 0 for ManagedNodeGroup

### ISSUE THREAD

**Issue by drunkirishcoder** [pulumi/pulumi-eks#645]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

`desiredSize` and `minSize` of 0 are not correctly handled when specifying the `NodeGroupScalingConfigArgs` for a `ManagedNodeGroup`.

### Steps to reproduce

1. Set either `desiredSize` or `minSize` to 0.
2. `pulumi up`
3. Check EKS management console and see `desiredSize` is 2 and `minSize` is 1

Expected: 

desiredSize = 0
minSize = 0

Actual: 

desiredSize = 2
minSize = 1

the offending code appears to be https://github.com/pulumi/pulumi-eks/blob/master/nodejs/eks/nodegroup.ts#L907-L908

```
const desiredSize = config && config.desiredSize || 2;
const minSize = config && config.minSize || 1;
```

because 0 is falsy, the expression `0 || 2` evaluates to `2` not `0`.

**Comment by leezen** [comment]
Thanks for pointing out the bugs -- any chance you'd be willing to submit a PR for this?

**Comment by drunkirishcoder** [comment]
submitted.

**Comment by stack72** [comment]
Closed via #654
